### ISSUE TITLE
VACUUM statement

### ISSUE THREAD

**Issue by mpf82** [tobymao/sqlglot#533]
``VACUUM`` is currently not supported. I tried to give it a shot, but soon realized, there are too many flavors of ``VACUUM`` across different dialects.

Postgres https://www.postgresql.org/docs/current/sql-vacuum.html

> VACUUM [ ( option [, ...] ) ] [ table_and_columns [, ...] ]
> VACUUM [ FULL ] [ FREEZE ] [ VERBOSE ] [ ANALYZE ] [ table_and_columns [, ...] ]

SQLITE https://www.sqlite.org/lang_vacuum.html

> VACUUM [schema] [INTO filename]

redshift https://docs.aws.amazon.com/redshift/latest/dg/r_VACUUM_command.html

> VACUUM [ FULL | SORT ONLY | DELETE ONLY | REINDEX | RECLUSTER ]  [ [ table_name ] [ TO threshold PERCENT ] [ BOOST ] ]

spark https://learn.microsoft.com/en-us/azure/databricks/spark/latest/spark-sql/language-manual/delta-vacuum

> VACUUM table_name [RETAIN num HOURS] [DRY RUN]

**Comment by barakalon** [comment]
Curious - how are you using sqlglot that you encounter vacuum expressions?

Are you trying to transpile them? Or just parse them?

**Comment by mpf82** [comment]
@barakalon I have integrated sqlglot into our company's database connection wrapper. The backend may vary, depeding on the software product and our user's environment, but mostly PostgreSQL, Oracle, sometimes SQLite (for tests or when running on devices that cannot access the network). 

Using sqlglot so far made the life of my colleagues somewhat easier, because they can focus on writing statements in the dialect they are comfortable with and do not have to worry that much about a dialects specifics (e.g. when using OFFSET / LIMIT) - even when told to stick to ANSI sql whenever possible.

Another bonus is the custom formatter I have derived from the sqlglot formatter (_pretty_ is just too much, IMHO), that logs the SQL statements in a standardized manner, no matter the "personal formatting touch" of individual programmers.

We are also using successfully parsed statements to write statistics, so we can easily detect bad performing statements (e.g. by extracting table names and columns used in WHERE and checking if we can gain performance by creating/altering indexes on those tables).

Of course there's still some statements, like VACUUM, that just cannot be transpiled, but I'd prefer if sqlglot could just parse those statements w/o throwing an error.

Statements that cannot be parsed are just executed as-is, but I've configured such cases to log in ERROR level, so I can try to create a PR myself, like I did when we encountered that [TO_TIMESTAMP](https://github.com/tobymao/sqlglot/pull/532) in Postgres can be either called with (double) or with (text, text).

In other cases, where I have no idea how to tackle them in sqlglot, I'd just create an issue here and hope for the best.

> Are you trying to transpile them? Or just parse them?

TL;DR: a little bit of both ;)
